<launch>

<!-- Set arguments -->
<arg name="haut_niv" default="False"/>


<!-- Run one lidar_front_data node -->
<node pkg="course_2023_pkg" name="get_front_data" type="lidar_data_reel.py" output="screen" required="true">
    <param name="front_topic" value="/front_data"/>
    <param name="side_topic" value="/side_data"/>
    <param name="lidar_topic" value="/LidarScan"/>
    <param name="angle0" value="130"/> <!--angles en degres, convertis en radians par la suite-->
    <param name="angle1" value="230"/> <!--angles en degres, convertis en radians par la suite-->

</node>

<!-- Run one lidar_data_process -->
<node pkg="course_2023_pkg" name="data_process" type="lidar_data_process.py" output="screen" required="true">
    <param name="front_data_topic" value="/front_data"/>
    <param name="side_data_topic" value="/side_data"/>
    <param name="angle0" value="130"/> <!--angles en degres, convertis en radians par la suite-->
    <param name="angle1" value="230"/> <!--angles en degres, convertis en radians par la suite-->
    <param name="FREE_SPACE_THRESH" value="0.6"/>
    

</node>

<!-- Run one lidar_nav_controller -->
<node pkg="course_2023_pkg" name="nav_controller" type="lidar_nav_controller.py" output="screen" required="true">
    
</node>

<param name="MAX_SPEED" value="1"/>
<param name="MAX_ANGLE" value="1"/>
<param name="SAFETY_DIST" value="0.85"/>
<param name="CLIP_DIST" value="3"/>
<param name="MIN_DIST" value="0.7"/>



<param name="lidar_nav_only" value="False"/>

<param name="kc" value="-0.85"/>
<param name="ks" value="1"/>
<!-- param lead controler-->
<param name="kd" value="6.5"/>
<param name="tau" value="1.0e-6"/>
<param name="a" value="10"/>


<param name="step_size" value="5"/>


<!--angles utilises pour calcul de front_dist-->
<param name="i0" value="165"/> 
<param name="i1" value="195"/> 

<param name="N_cadrans" value='5'/>


<param name="haut_niv" value="$(arg haut_niv)"/>


</launch>